osrparse/replay.py

```diff
@@ -88,15 +89,14 @@ def parse_replay_data(replay_data_str, mode):
             if time_delta == -12345 and event == events[-1]:
                 rng_seed = keys
                 continue
-
-            if mode is GameMode.STD:
+            if mode == GameMode.STD:
                 keys = Key(keys)
                 event = ReplayEventOsu(time_delta, float(x), float(y), keys)
-            if mode is GameMode.TAIKO:
+            if mode == GameMode.TAIKO:
                 event = ReplayEventTaiko(time_delta, int(x), KeyTaiko(keys))
-            if mode is GameMode.CTB:
+            if mode == GameMode.CTB:
                 event = ReplayEventCatch(time_delta, float(x), int(keys) == 1)
-            if mode is GameMode.MANIA:
+            if mode == GameMode.MANIA:
                 event = ReplayEventMania(time_delta, KeyMania(int(x)))
 
```

Comment:
Does the rust parser break `is` checks? `is` is the recommended way to compare enums in python, so this would have to be a breaking change if we're unable to replicate it in rust. Would much prefer (almost to the point of it being a requirement) if `is` checks worked as-is.